### PR TITLE
[25.0 backport] vendor: OTEL v0.46.1 / v1.21.0

### PR DESCRIPTION
**Pull request by austinvazquez**
**- What I did**
Backports https://github.com/moby/moby/pull/47245 to 25.0

**- How I did it**
```
git cherry-pick -xsS c516804d6f13f7bdd337748ff52563d863e3cfaa
bash hack/vendor.sh all
```

**- How to verify it**
CI must be successful

**- Description for the changelog**
```markdown changelog
Vendor OTEL v0.46.1 / v1.21.0
```

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by austinvazquez** [comment]
Can we have a rerun for test / validate? Local validate swagger-gen is okay.
```
...
docker run --rm --privileged  -e BUILDFLAGS -e KEEPBUNDLE -e DOCKER_BUILD_ARGS -e DOCKER_BUILD_GOGC -e DOCKER_BUILD_OPTS -e DOCKER_BUILD_PKGS -e DOCKER_BUILDKIT -e DOCKER_BASH_COMPLETION_PATH -e DOCKER_CLI_PATH -e DOCKERCLI_VERSION -e DOCKERCLI_REPOSITORY -e DOCKERCLI_INTEGRATION_VERSION -e DOCKERCLI_INTEGRATION_REPOSITORY -e DOCKER_DEBUG -e DOCKER_EXPERIMENTAL -e DOCKER_GITCOMMIT -e DOCKER_GRAPHDRIVER -e DOCKER_LDFLAGS -e DOCKER_PORT -e DOCKER_REMAP_ROOT -e DOCKER_ROOTLESS -e DOCKER_STORAGE_OPTS -e DOCKER_TEST_HOST -e DOCKER_USERLANDPROXY -e DOCKERD_ARGS -e DELVE_PORT -e GITHUB_ACTIONS -e TEST_FORCE_VALIDATE -e TEST_INTEGRATION_DIR -e TEST_INTEGRATION_USE_SNAPSHOTTER -e TEST_INTEGRATION_FAIL_FAST -e TEST_SKIP_INTEGRATION -e TEST_SKIP_INTEGRATION_CLI -e TEST_IGNORE_CGROUP_CHECK -e TESTCOVERAGE -e TESTDEBUG -e TESTDIRS -e TESTFLAGS -e TESTFLAGS_INTEGRATION -e TESTFLAGS_INTEGRATION_CLI -e TEST_FILTER -e TIMEOUT -e VALIDATE_REPO -e VALIDATE_BRANCH -e VALIDATE_ORIGIN_BRANCH -e VERSION -e PLATFORM -e DEFAULT_PRODUCT_LICENSE -e PRODUCT -e PACKAGER_NAME -e PAGER -e GIT_PAGER -e OTEL_EXPORTER_OTLP_ENDPOINT -e OTEL_EXPORTER_OTLP_PROTOCOL -e OTEL_SERVICE_NAME -v "/home/ec2-user/moby/bundles:/go/src/github.com/docker/docker/bundles" -v "/home/ec2-user/moby/.git:/go/src/github.com/docker/docker/.git" -v docker-dev-cache:/root/.cache -v docker-mod-cache:/go/pkg/mod/     -t -i "docker-dev" hack/validate/swagger-gen
Congratulations!  All API changes are done the right way.
[moby]$ 
```

**Comment by austinvazquez** [comment]
Nice CI looks good! cc: @thaJeztah , @corhere PTAL.

**Review by estesp — APPROVED** [review]

**Review by cpuguy83 — APPROVED** [review]

**Comment by cpuguy83** [comment]
@corhere I see you added yourself as a reviewer, I'll hold off merge until you approve.

File: api/server/router/grpc/grpc.go
Comment (thaJeztah) [inline api/server/router/grpc/grpc.go:24]: FWIW; PR that removes the deprecated calls is https://github.com/moby/moby/pull/47919

**Review by corhere — APPROVED** [review]

File: api/server/router/grpc/grpc.go
Comment (austinvazquez) [reply api/server/router/grpc/grpc.go:24]: Oh nice, although that might require a buildkit upgrade to v0.14.0 for some of the utility package updates for tracing.

Comment (thaJeztah) [reply api/server/router/grpc/grpc.go:24]: Oh! Good one; didn't remember that those may be needed (or at least, I didn't check if those were _required_ for that patch, or if there's alternatives).

In either case; the deprecated ones still work (at least for now) so not critical for sure!

**Pull request merged**
